CONTRIBUTING.md

```diff
@@ -0,0 +1,250 @@
+
+Since the `ui-kit` is using typescript please make sure to define proper prop types for your components. This will improve the documentation, which creates dynamically and will provide better error messages for the users. And also this will provide auto-completion in some IDE’s. Describe the props as good as possible, so also define object shapes or array shapes.
+
+#### Never hard code HTML ID’s
+
+This might conflict at some point. If you need an id on the element expose a prop for that so the user can define it based on the application.
+
+#### Use Logical Defaults
+
+Use prop defaults which make sense this will reduce the efforts needed to adopt this component.
+
+#### A11y Support
+
+A good start on that topic is to read [this website](https://a11yproject.com/)
+While creating your component think about users: - Which uses keyboards - expose tab indexes as props - Use semantic correct HTML - ARIA
+
+#### Configuration Objects
+
+Please consider using configuration objects as they will provide the user a lot of value by
+
+* consistency
+* less typing needed
+* less error prone
+
+But please avoid using configuration Objects if you have 3 or fewer properties.
+
+#### Single Responsibility Principle
+
+Keep the components simple by applying them to a single responsibility. In that way, the component will stay simple in most cases. Try not to introduce unnecessary props. Each prop is a configuration which will automatically increase the complexity. Before adding a property to an existing component ask yourself if this could be a new component.
+
+#### Keep Components Simple
+
+It's important to keep components as simple as possible, add logic to a component if it improves performance or quality of that component. For example, paginating a list if the number of items is exceeding a certain amount or having a show toggle to show more content like a dropout.
+
+#### Directory Structure
+
+Components get grouped into `packages`. The Strucutre looks like this:
+
+```
+packages
+└── <ComponentGroup>
+    ├── README.md
+    ├── components
+    │   └── <Component>.tsx
+    ├── index.ts
+    ├── stories
+    │   └── <ComponentGroup>.stories.tsx
+    ├── style.ts
+    └── tests
+        └── <Component>.test.tsx
+```
+
+## Atomic Design
+
+Atomic design is the principle we are trying to follow while we create components. More information on the principle is available at [atomicdesign.bradfrost.com](http://atomicdesign.bradfrost.com/)
+
+### Atoms
+
+Atoms are the smallest building blocks and the form the foundation of the library. Atoms get composed together with other atoms to create a molecule.
+Some best practices here are:
+
+* Honour the native API’s, for example, if you define a prop for `tabIndex` call it `tabIndex` instead of `tabindex` or `indexOfTabs`. This will improve consistency.
+* Pass props via spread, if you need to pass props to a component using the spread operator like this `<input type="password" {...this.props} />`
```

Comment:
That could lead to non-html props ending up in html tags

Reply:
Hmm How do you think I can clarify that we should make sure only to use html attributes? in that example I was thinking about having `value`, `disabled`, `placeholder`, `tabIndex` what we could do is 
```JS
function(props) {
  const {tabIndex, value, disabled, placeholder} = props;
  const attributes = {tabIndex, value, disabled, placeholder};
  return <input type="password" {...attributes} />;
```

Also if we have mixed props we could give this example:

```JS
function(props) {
  const {not, html, props, ...rest} = props;
  return <input type="password" {...rest} />;
}
```

What do you think @nLight